src/Interpreters/ConcurrentHashJoin.h

```diff
@@ -1,40 +1,27 @@
 #pragma once
-
-#include <condition_variable>
 #include <memory>
-#include <optional>
-#include <Functions/FunctionsLogical.h>
-#include <Interpreters/Context.h>
-#include <Interpreters/ExpressionActions.h>
-#include <Interpreters/HashJoin.h>
 #include <Interpreters/IJoin.h>
-#include <base/defines.h>
-#include <base/types.h>
-#include <Common/Stopwatch.h>
-
+#include <Interpreters/Context_fwd.h>
+#include <Interpreters/HashJoin.h>
+#include "Context.h"
 namespace DB
 {
 
-/**
```

Comment:
Why have you deleted the algorithm description? If the behavior is changed within this PR, please update the comment accordingly